From: SeongJae Park <sj@kernel.org>
Date: Mon, 7 Oct 2024 09:42:17 -0700
Subject: [PATCH] ==== remove DAMON debugfs interface ====

[RFC PATCH] mm/damon: remove DAMON debugfs interface

DAMON debugfs interface was the only user interface at the beginning[1].
After it has merged into the mainline, it became clear that it is not
flexible.

In 2022-02[2], we therefore introduced DAMON sysfs interface as another
user interface aiming long-term flexibility and stability.  With its
introduction, DAMON debugfs interface was announced to be deprecated in
near future.

In 2023-02[3], we marked DAMON debugfs interface as officially
deprecated.  In 2024-01[4], we made the marking difficult to be
mistakenly ignored.

So about three years since the deprecation plan announcement, and about
two years since the official deprecation have passed.  It is believed to
be long time line that enough to be found by normal users.  And no
serious concern from users has reported so far.  This this must be the
not-to-fast time to say good-bye to DAMON debugfs interface.

Now finally remove the DAMON debugfs interface code from the tree with
this RFC patch series.  The RFC tag would be dropped after next LTS
kernel is announced (hopefully by the end of 2024), and we get no
serious concern about this change by then.

[1] https://lore.kernel.org/20210716081449.22187-1-sj38.park@gmail.com
[2] https://lore.kernel.org/20220228081314.5770-1-sj@kernel.org
[3] https://lore.kernel.org/20230209192009.7885-1-sj@kernel.org
[4] https://lore.kernel.org/20240130013549.89538-1-sj@kernel.org

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/MmeuEpnX | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/MmeuEpnX

diff --git a/damon_meta_changes/MmeuEpnX b/damon_meta_changes/MmeuEpnX
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

